tools/RAiDER/models/credentials.py

```diff
@@ -0,0 +1,90 @@
+'''
+dict that hold api information and help url for downloading weather model data
+   -  api information is stored in hidden file in home directory 
+
+        api filename      weather models          uid           key
+        cdsapirc          ERA5, ERA5T             uid           key
+        ecmwfapirc        ERAI, HRES              email         key
+        netrc             GMAO, MERRA2            username      password
+        <NAN>             HRRR [public access]    <NAN>         <NAN> 
+'''
+
+MODEL_API_DICT = {
+        'cdsapirc' :   {'api' : """\
+                                url: https://cds.climate.copernicus.eu/api/v2\
+                                \nkey: {uid}:{key}
+                                """,
+                        'url' : 'https://cds.climate.copernicus.eu/api-how-to'
+                        },
+        'ecmwfapirc' : {'api' : """{{\
+                                 \n"url"   : "https://api.ecmwf.int/v1",\
+                                 \n"key"   : "{key}",\
+                                 \n"email" : "{uid}"\
+                                 \n}}
+                                """,
+                        'url' : 'https://confluence.ecmwf.int/display/WEBAPI/Access+ECMWF+Public+Datasets#AccessECMWFPublicDatasets-key'
+                        },
+        'netrc' :       {'api' : """\
+                                \nmachine urs.earthdata.nasa.gov\
+                                \n        login {uid}\
+                                \n        password {key}\
+                                """,
+                       'url': 'https://wiki.earthdata.nasa.gov/display/EL/How+To+Access+Data+With+cURL+And+Wget'
+                       }
+                }
+
+# Function to check and write MODEL API credential for downloading weather model data
+def check_api(model: str,
+              UID: str = None,
+              KEY: str = None,
+              prompt_flag: bool = True,
+              update_flag: bool = False) -> None:
+    import os
+
+    # Weather model credential filename
+    # typically stored in home dir as hidden file
+    if model in ('ERA5', 'ERA5T'):
+        ap_filename = 'cdsapirc'
+    elif model in ('ERAI, HRES'):
+        api_filename = 'ecmwfapirc'
+    elif model in ('GMAO'):
+        api_filename = 'netrc'
+    else:
+        api_filename = None #for HRRR
+
+    if api_filename:    
+        # Check if the credential api file exists
+        api_filename_path = os.path.expanduser('~/.')+ f'{api_filename}'
```

Comment:
pathlib?

You can then do things like `Path('~/.') / api_filename`

Reply:
Does hardcoding the path with forward slash make this only suitable for linux?

Reply:
I found here the same way of cross-platform dealing with homedir:
https://stackoverflow.com/questions/4028904/what-is-a-cross-platform-way-to-get-the-home-directory 
But open to suggestions.

Reply:
@cmarshak  I also prefer pathlib, but did not find it in environmental.yaml . So avoided introducing a dependency on 
 additional module and used the default one "os"

Reply:
@mgovorcin `pathlib` is in the python standard library, same as `os`:
https://docs.python.org/3/library/pathlib.html

Reply:
@jhkennedy thanks, then I will use pathlib instead os